### DISCUSSION TITLE
MongoDB Outbox Pattern Implementation

### DISCUSSION THREAD

**Discussion by cookie-bytes** [BEagle1984/silverback#189] [Ideas]
@BEagle1984 just wondering if you would be interested in a PR for MongoDB implementation for the Outbox pattern?
We are looking to implement this ourselves and thought it worth asking if makes sense to add it to the Silverback packages.
Thanks

**Comment by BEagle1984** [comment]
I'd be very interested in such a contribution. 👍 

Beware that I've been working on the next big release for quite some time and the database part is being rewritten from scratch. The new approach will lead to a dedicated package per database engine since I wanted to get rid of the dependency on EF (to allow the integration with other ORMs or data access strategies, etc.) and optimize it a bit more (getting rid of the sub-optimal distributed lock mechanism as well).
I'm still working on the different features and I implemented SQLite only at the moment, as a sort of showcase: https://github.com/BEagle1984/silverback/tree/release/5.0.0/src/Silverback.Storage.Sqlite/Storage.
I plan for sure to be supporting SQL Server, Postgres, and MongoDB as storage and offer other possibilities for locking (like Kubernetes leader election, still to be defined).

So, depending on the urgency and your bravery, you could decide to wait and implement it in the new version.
I can't, unfortunately, give an estimation for the delivery of an MVP as there are quite a few features I'd like to include.
